core/fslock/fslock.cpp

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "fslock/fslock.hpp"
+
+#include <boost/filesystem/fstream.hpp>
+#include <boost/filesystem/operations.hpp>
+#include "fslock/fslock_error.hpp"
+
+namespace fc::fslock {
+  // TODO(artyom-yurin): Should be unlocked if process died
+  outcome::result<boost::interprocess::file_lock> lock(
+      const std::string &file_lock_path) {
+    try {
+      if (!boost::filesystem::exists(file_lock_path)) {
+        boost::filesystem::ofstream os(file_lock_path);
+        os.close();
+      }
+      boost::interprocess::file_lock file_lock(file_lock_path.c_str());
+      if (!file_lock.try_lock()) {
+        return FSLockError::FILE_LOCKED;
```

Comment:
So this is a client's responsibility to interpret and handle locked paths, isn't it?

Reply:
Not really. This tool will be used to block repositories, if the file could not be locked, then the repository will not be locked.